File: java/resistor-color-duo/src/main/java/ResistorColorDuo.java
import java.util.List;

class ResistorColorDuo {

    List<String> resColors = List.of(
        "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white"
    );

    int value(String[] colors) {

        if (colors.length < 2) { throw new IllegalArgumentException(); }

        return resColors.indexOf(colors[0]) * 10 + resColors.indexOf(colors[1]);
    }
}

File: csharp/resistor-color-duo/ResistorColorDuo.cs
using System;

public static class ResistorColorDuo
{
    private static readonly string[] _colors =
        { "black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white" };

    public static int Value(string[] colors)
    {
        return Array.IndexOf(_colors, colors[0]) * 10 +
               Array.IndexOf(_colors, colors[1]);

    }
}
